### PR TITLE
Integrated VS Emulator for Android.

### PR DESCRIPTION
**Pull request by dlebu**
Added the option to launch VS Emulator for Android, if the user has it installed. Usage:
>`taco emulate --vsemulator`

The emulator is available for install [here](https://www.visualstudio.com/en-us/features/msft-android-emulator-vs.aspx).

@subhagpo @nisheetjain @MSLaguana @joshuaskelton @guillaumejenkins @meenakb @digeff

### CODE REVIEW

File: src/taco-cli/cli/utils/vsEmulator.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+import resources = require("../../resources/resourceManager");
+import utils = require("taco-utils");
+import Logger = utils.Logger;
+
+/**
+ * Handles the Visual Studio Emulator for Android specific operations.
+ */
+export class VSEmulator {
+
+    private static CLI_PATH: string = process.env["ProgramFiles(x86)"] + "\\Microsoft Emulator Manager\\1.0\\emulatorcmd.exe";
```
Comment (nisheetjain) [inline src/taco-cli/cli/utils/vsEmulator.ts]: what about 32-bit?

File: src/taco-cli/cli/utils/vsEmulator.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        return fs.existsSync(path.resolve(VSEmulator.CLI_PATH));
+    }
+
+    /**
+     * Checks if the emulator is correctly set-up and has installed platforms.
+     */
+    public static isReady(): Q.Promise<boolean> {
+        if (!VSEmulator.isInstalled()) {
+            return Q.resolve(false);
+        } else {
```
Comment (nisheetjain) [inline src/taco-cli/cli/utils/vsEmulator.ts]: else not needed

Comment (joshuaskelton) [inline src/taco-cli/cli/utils/vsEmulator.ts]: Is this just a style nit? Is the below preferred?
```javascript
public static foo(arg: Boolean): string {
   if (arg) {
      return "Only if";
   }
   return "No else";
}
```

File: src/taco-cli/cli/emulate.ts [hunk 3]
```diff
@@ -129,13 +133,64 @@ class Emulate extends commands.TacoCommandBase {
         return Q.all<any>([PlatformHelper.determinePlatform(commandData), Settings.loadSettingsOrReturnEmpty()])
             .spread((platforms: PlatformHelper.IPlatformWithLocation[], settings: Settings.ISettings): Q.Promise<any> => {
                 buildTelemetryHelper.storePlatforms(telemetryProperties, "actuallyBuilt", platforms, settings);
+
                 return PlatformHelper.operateOnPlatforms(platforms,
-                    (localPlatforms: string[]): Q.Promise<any> => CordovaWrapper.emulate(commandData, localPlatforms),
+                    (localPlatforms: string[]): Q.Promise<any> =>
+                        Emulate.handleVsEmulator(commandData, localPlatforms)
```
Comment (nisheetjain) [inline src/taco-cli/cli/emulate.ts]: may be we can use PlatformHelper.determinePlatform here

File: src/taco-cli/cli/utils/vsEmulator.ts (hunk 1 above)
Comment (dlebu) [inline src/taco-cli/cli/utils/vsEmulator.ts]: Per our discussion, I will add 32-bit support if we decide to merge this to master.

File: src/taco-cli/cli/utils/vsEmulator.ts (hunk 2 above)
Comment (dlebu) [inline src/taco-cli/cli/utils/vsEmulator.ts]: I agree it is not needed, but I feel the code is more readable if we keep it unchanged.

File: src/taco-cli/cli/emulate.ts (hunk 3 above)
Comment (dlebu) [inline src/taco-cli/cli/emulate.ts]: Definitely. I will add that when we merge this MVP to master, per our discussion today.

**Comment by dlebu** [comment]
Merging this PR since @subhagpo will try to get feedback for it soon.

**Pull request merged**